Crashing when set date/location

Happens from SkySafari with Nexstar with "set date/location" feature enabled.

```
./dist/alpacascope-0.0.6-darwin-amd64
INFO[0000] Looking for Alpaca Remote Server locally on port 11111...
INFO[0000] No local Alpaca Remote Servers found
INFO[0000] Discovered Alpaca Server on 172.16.1.115:11111
INFO[0000] Starting SkyFi Discovery service on UDP/4031
INFO[0000] Connected to telescope 0: CPWI_ASCOM_Telescope
INFO[0000] Waiting for nexstar clients on 0.0.0.0:4030
ERRO[0033] Error talking to scope: 1024: Property write UTCDate is not implemented in this driver.
panic: runtime error: index out of range [7] with length 7

goroutine 21 [running]:
main.NexstarToLatLong(...)
	/Users/aturner/go/src/github.com/synfinatic/alpacascope/cmd/nexstar.go:313
main.nexstar_command(0xc00000e6a0, 0x9, 0xc0000b3000, 0x400, 0x400, 0x0, 0x0, 0x8)
	/Users/aturner/go/src/github.com/synfinatic/alpacascope/cmd/nexstar.go:175 +0x1813
main.handleNexstar(0x13ef500, 0xc00009c070, 0xc00000e6a0)
	/Users/aturner/go/src/github.com/synfinatic/alpacascope/cmd/nexstar.go:24 +0x14c
created by main.main
	/Users/aturner/go/src/github.com/synfinatic/alpacascope/cmd/main.go:170 +0xb05
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,8 @@
 Added:
 
  - Support for Nexstar tracking commands: t and T
+ - Fix crash when setting Lat/Long due to index out of range
+ - Clean up Nexstar cmd processing
 
 ## v0.0.5 - 2021-01-17
 

--- a/cmd/nexstar.go
+++ b/cmd/nexstar.go
@@ -62,61 +62,55 @@ func nexstar_command(t *alpaca.Telescope, len int, buf []byte) []byte {
 		ra, dec, err := t.GetRaDec()
 		if err != nil {
 			log.Errorf("Unable to get RA/DEC: %s", err.Error())
-			break
-		}
-		radec := telescope.Coordinates{
-			RA:  ra,
-			Dec: dec,
-		}
+		} else {
+			radec := telescope.Coordinates{
+				RA:  ra,
+				Dec: dec,
+			}
 
-		high_precision := true
-		if buf[0] == 'E' {
-			high_precision = false
+			high_precision := true
+			if buf[0] == 'E' {
+				high_precision = false
+			}
+			ret = fmt.Sprintf("%s#", radec.Nexstar(high_precision))
 		}
-		ret = fmt.Sprintf("%s#", radec.Nexstar(high_precision))
 
 	case 'Z':
 		// Get AZM/ALT.  Note that AZM is 0->360, while Alt is -90->90
 		azm, alt, err := t.GetAzmAlt()
 		if err != nil {
 			log.Errorf("Unable to get AZM/ALT: %s", err.Error())
-			break
+		} else {
+			azm_int := uint32(azm / 360.0 * math.Pow(2, 16))
+			alt_int := uint32(alt / 360.0 * math.Pow(2, 16))
+			ret = fmt.Sprintf("%04X,%04X#", azm_int, alt_int)
 		}
-
-		azm_int := uint32(azm / 360.0 * math.Pow(2, 16))
-		alt_int := uint32(alt / 360.0 * math.Pow(2, 16))
-		ret = fmt.Sprintf("%04X,%04X#", azm_int, alt_int)
 
 	case 'z':
 		// Get Precise AZM/ALT
 		azm, alt, err := t.GetAzmAlt()
 		if err != nil {
 			log.Errorf("Unable to get AZM/ALT: %s", err.Error())
-			break
+		} else {
+			azm_int := uint32(azm / 360.0 * 4294967296.0)
+			alt_int := uint32(alt / 360.0 * 4294967296.0)
+			ret = fmt.Sprintf("%08X,%08X#", azm_int, alt_int)
 		}
-
-		azm_int := uint32(azm / 360.0 * 4294967296.0)
-		alt_int := uint32(alt / 360.0 * 4294967296.0)
-		ret = fmt.Sprintf("%08X,%08X#", azm_int, alt_int)
 
 	case 't':
 		// get tracking mode
 		mode, err := t.GetTracking()
 		if err != nil {
 			log.Errorf("Unable to get tracking mode: %s", err.Error())
-			break
+		} else {
+			ret = fmt.Sprintf("%d#", mode)
 		}
-		ret = fmt.Sprintf("%d#", mode)
 
 	case 'T':
 		// set tracking mode
 		var tracking_mode alpaca.TrackingMode
 		fmt.Sscanf(string(buf[1]), "%d", &tracking_mode)
-		err := t.PutTracking(tracking_mode)
-		if err != nil {
-			log.Errorf("Unable to set tracking mode: %s", err.Error())
-			break
-		}
+		err = t.PutTracking(tracking_mode)
 		ret = "#"
 
 	case 'V':
@@ -126,10 +120,6 @@ func nexstar_command(t *alpaca.Telescope, len int, buf []byte) []byte {
 	case 'P':
 		// Slew
 		err = executeSlew(t, buf)
-		if err != nil {
-			log.Errorf("Unable to slew: %s", err.Error())
-			break
-		}
 		ret = "#"
 
 	case 's':
@@ -158,31 +148,33 @@ func nexstar_command(t *alpaca.Telescope, len int, buf []byte) []byte {
 
 	case 'w':
 		// get location
+		failed := false
 		lat, err := t.GetSiteLatitude()
 		if err != nil {
-			break
+			log.Errorf("Error talking to scope: %s", err.Error())
+			failed = true
 		}
 
 		long, err := t.GetSiteLatitude()
 		if err != nil {
-			break
+			// logged at the end
+			failed = true
 		}
-		ret_val = LatLongToNexstar(lat, long)
-		ret_val = append(ret_val, '#')
+
+		if !failed {
+			ret_val = LatLongToNexstar(lat, long)
+			ret_val = append(ret_val, '#')
+		}
 
 	case 'W':
 		// set location
-		lat, long := NexstarToLatLong(buf[1:8])
+		lat, long := NexstarToLatLong(buf[1:9])
 		err = t.PutSiteLatitude(lat)
 		if err != nil {
-			log.Errorf("Error setting site latitude: %s", err.Error())
-			break
+			log.Errorf("Error talking to scope: %s", err.Error())
 		}
 		err = t.PutSiteLongitude(long)
-		if err != nil {
-			log.Errorf("Error setting site longitude: %s", err.Error())
-			break
-		}
+		// logged at the end
 		ret = "#"
 
 	case 'H':

